cmd/outline-ss-server/main.go

```diff
@@ -287,6 +331,56 @@ func (s *OutlineServer) runConfig(config Config) (func() error, error) {
 							return serviceConfig.Dialer.Fwmark
 						}())
 						go service.PacketServe(pc, ssService.NewPacketAssociation, s.serverMetrics)
+					case listenerTypeWebsocketStream:
+						if _, exists := webServers[lnConfig.WebServer]; !exists {
+							return fmt.Errorf("listener type `%s` references unknown web server `%s`", lnConfig.Type, lnConfig.WebServer)
+						}
+						mux := webServers[lnConfig.WebServer]
+						handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+							handler := func(wsConn *websocket.Conn) {
+								defer wsConn.Close()
+								ctx, contextCancel := context.WithCancel(context.Background())
+								defer contextCancel()
+								raddr, err := net.ResolveTCPAddr("tcp", r.RemoteAddr)
+								if err != nil {
+									slog.Error("failed to upgrade", "err", err)
+									w.WriteHeader(http.StatusBadGateway)
+									return
+								}
+								conn := &streamConn{&wrappedConn{Conn: wsConn, raddr: raddr}}
+								ssService.HandleStream(ctx, conn)
+							}
+							websocket.Handler(handler).ServeHTTP(w, r)
+						})
+						mux.Handle(lnConfig.Path, http.StripPrefix(lnConfig.Path, handler))
+					case listenerTypeWebsocketPacket:
+						if _, exists := webServers[lnConfig.WebServer]; !exists {
+							return fmt.Errorf("listener type `%s` references unknown web server `%s`", lnConfig.Type, lnConfig.WebServer)
+						}
+						mux := webServers[lnConfig.WebServer]
+						handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
This is ok for this PR, but it doesn't work with half-closed. We need to migrate to the Gorilla library to support that.